controllers/users/populateUser.js

```diff
@@ -10,6 +11,14 @@ module.exports = async (req, res) => {
       filterAllowDm = `AND users.allow_anon_dm = ${allow_anon_dm}`;
     }
 
+    const blockedIds = await UsersFunction.getBlockedAndBlockerUserId(UserBlockedUser, req.userId);
+    let filterBlockedUser = '';
+    if (blockedIds.length > 0) {
+      filterBlockedUser += `AND users.user_id NOT IN (${blockedIds
+        .map((item) => `'${item}'`)
+        .join(',')})`;
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code here is constructing a SQL query string using JavaScript string concatenation. This could potentially lead to SQL injection if the `blockedIds` are not properly sanitized. Although it seems like the `blockedIds` are coming from a trusted source (the `getBlockedAndBlockerUserId` function), it's still a good practice to use parameterized queries or prepared statements to prevent any potential SQL injection.

```diff
- filterBlockedUser += `AND users.user_id NOT IN (${blockedIds
-   .map((item) => `'${item}'`)
-   .join(',')})`;
+ const blockedIdsPlaceholders = blockedIds.map((_, index) => `$${index + 1}`).join(',');
+ filterBlockedUser += `AND users.user_id NOT IN (${blockedIdsPlaceholders})`;
```
In your SQL query, you can then use `sequelize.query` with replacements:

```javascript
let queryResults = await sequelize.query(
  `SELECT 
    user_id,
    // ...
    ${filterBlockedUser}
  `,
  {
    replacements: [...blockedIds],
    type: sequelize.QueryTypes.SELECT,
  }
);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->